src/did/Did.utils.ts

```diff
@@ -45,3 +57,94 @@ export function getAddressFromIdentifier(
   }
   return identifier.substr(IDENTIFIER_PREFIX.length)
 }
+
+export function createDefaultDidDocument(
+  identifier: string,
+  publicBoxKey: string,
+  publicSigningKey: string,
+  kiltServiceEndpoint?: string
+): IDidDocument {
+  return {
+    id: identifier,
+    '@context': CONTEXT,
+    authentication: {
+      type: KEY_TYPE_AUTHENTICATION,
+      publicKey: [`${identifier}#key-1`],
+    },
+    publicKey: [
+      {
+        id: `${identifier}#key-1`,
+        type: KEY_TYPE_SIGNATURE,
+        controller: identifier,
+        publicKeyHex: publicSigningKey,
+      },
+      {
+        id: `${identifier}#key-2`,
+        type: KEY_TYPE_ENCRYPTION,
+        controller: identifier,
+        publicKeyHex: publicBoxKey,
+      },
+    ],
+    service: kiltServiceEndpoint
+      ? [
+          {
+            type: SERVICE_KILT_MESSAGING,
+            serviceEndpoint: kiltServiceEndpoint,
+          },
+        ]
+      : [],
+  }
+}
+
+export function verifyDidDocumentSignature(
+  didDocument: IDidDocumentSigned,
+  address: string
```

Comment:
Can we have the did identifier here, check the did identifier against the one in the did document and use it for checking the signature?

Reply:
I've updated this, it's ready for re-review!

Reply:
sounds good, done 👍